Leaflet-finder/leaflet-dask-array.py

```diff
@@ -28,21 +16,22 @@
 print (type (Scheduler_IP))
 #print (Client(Scheduler_IP))
 c = Client(Scheduler_IP)
-print (c.get_versions(check=True))
-print(sys.path)
+#print (c.get_versions(check=True))
+#print(sys.path)
 
-def Leaflet_finder(block, atoms, cutoff, len_chunks, block_id=None):
+def Leaflet_finder(matrix, atoms, cutoff, len_chunks,dtype=None,block_id=None):
+    print(atoms, cutoff, len_chunks, block_id)
     id_0 = block_id[0]
     id_1 = block_id[1]
 
 #     print(len_chunks, len(atoms))
-    block[:,:] = cdist(atoms[id_0*len_chunks:(id_0+1)*len_chunks], atoms[id_1*len_chunks:(id_1+1)*len_chunks]) <= cutoff 
-    S = scipy.sparse.dok_matrix((len(atoms), len(atoms)))
-    S[id_0*len_chunks:(id_0+1)*len_chunks, id_1*len_chunks:(id_1+1)*len_chunks] = block
-    l = np.array({i: item for i, item 
+    matrix = cdist(atoms[id_0*len_chunks:(id_0+1)*len_chunks], atoms[id_1*len_chunks:(id_1+1)*len_chunks]) <= cutoff
```

Comment:
You pass `matrix` into the function but now you are overwriting it; if you wanted to store the distances then it should be
```
matrix[:, :] = ...
```
(unless I misunderstand how `dask.array.map_blocks()` works.

However, I don't think that we really need it. You (or rather @mkhoshle )  are actually using `da.map_blocks(Leaflet_finder, ...)` just as a way to generate tasks with indices corresponding to the blocks.  

Wouldn't it be cleaner to then just write it with `delayed`?